Automattic/wp-calypso
# Storage add-ons v2: Expose the extended storage add-ons behind the feature flag.

Related to #97221

## Proposed Changes

This PR exposes the extended storage add-ons to the client side behind the feature flag, `upgrades/storage-add-on-v2`. The flag will be enabled by default in the development environment.

`/add-ons` with the feature flag on:

![CleanShot 2024-12-26 at 14 33 34@2x](https://github.com/user-attachments/assets/442a3c75-31ab-4a92-8439-e5273c2a344d)

`/plans` with the feature flag on:

![CleanShot 2024-12-26 at 16 14 58@2x](https://github.com/user-attachments/assets/3a61b666-1343-4976-99aa-b1d49b604a73)


**Caution:**
* It's known issue that a Business site can't purchase the add-ons over 200G. It will be fixed from the backend.
* Another known issue is that the total spaces shown in the storage dropdown are wrong for a site owning a storage add-on. This will also be fixed as a follow-up. 

## Why are these changes being made?


For establishing the foundation of exposing the extended storage add-ons, i.e. 150G, 200G, 250G, 300G and 350G. It is feature-gated for now so we will be able to continuously deploy improvements without disturbing the production experience.

## Testing Instructions



* Without the feature flag in any environment except for the development, the extended storage add-ons are not available in `/add-ons` or the storage dropdown in `/plans`.
* Enabling the feature flag by appending `?flags=upgrades/storage-add-on-v2` , the extended storage add-ons should be available in `/add-ons` and the storage dropdown in `/plans`.


## Pre-merge Checklist
